src/bitops.c

```diff
@@ -888,7 +888,7 @@ void bitposCommand(client *c) {
         addReplyLongLong(c, -1);
     } else {
         long bytes = end-start+1;
-        long pos = redisBitpos(p+start,bytes,bit);
+        long long pos = redisBitpos(p+start,bytes,bit);
```

Comment:
you also need to cast `bytes*8` and `start*8` below to avoid overflow in the multiplication.

Reply:
I miss it. Fix and add a test.
BTW there is no way to construct a bitmap of all 1. Or I can add more tests.

Reply:
i don't understand what you mean.

Reply:
I want to test bitcount overflow. It needs a sting 536870913 bytes with all bytes set to 255. But I can't generate such a string quickly.

Reply:
only way i can think of is a loop that uses SETRANGE with some big bulk of `\xff\xff\xff`.
i.e. you can generate a 1mb string in tcl, and then send 512 SETRANGE commands

Reply:
```
        set mega [expr (1 << 20)]
        set part [string repeat "\xFF" $mega]
        for {set i 0} {$i < 512} {incr i} {
            r setrange mykey [expr $i * $mega] $part
        }
```

This code takes more than 25 seconds. So I think we can give up.

Reply:
we can make it depend on `--accurate` i.e. `if {$::accurate}`

Reply:
OK. Test added.